### PR TITLE
[CSL-1840] Add devnet configuration key

### PR DESCRIPTION
**Pull request by gromakovsky**

### CODE REVIEW

File: node/configuration.yaml [hunk 1]
```diff
@@ -14945,6 +14945,51 @@ mainnet_launch_base:
...
+        protocolConstants:
+          <<: *mainnet_base_protocolConstants
+          protocolMagic: 170987650
+          k: 6 # set to desired value
+        avvmDistr: *mainnet_avvmDistr
+        initializer:
+          testnetInitializer:
+            testBalance:
+              poors:        200 # should be enough for developers
+              richmen:      4   # this is the number of core nodes,
```
Comment (deepfire) [inline node/configuration.yaml]: @gromakovsky, can we reduce that to `3`, to make the dev clusters lighter?

Comment (gromakovsky) [reply node/configuration.yaml]: I've set it to 3, please re-approve.

**Comment by deepfire** [comment]
@gromakovsky, it appears that I'm getting an error with commit `701c43e93c6a174bec4eae99ec6c7a0f334de594`:
```
cardano-keygen: ConfigurationParseFailure "/nix/store/prxks0x07l544idv0ah47fagx4m2wr9v-cardano-sl-701c43e/node/configuration.yaml" (AesonException "Error in $.devnet.node: key \"defaultPeers\" not present")
```

**Comment by gromakovsky** [comment]
Sorry, my bad, will fix shortly.

**Comment by gromakovsky** [comment]
@deepfire fixed. It's strange that tests didn't fail.

**Comment by deepfire** [comment]
Maybe it's due to laziness -- the piece of configuration that defines that `devnet` key never gets forced, because it's not exercised..

**Comment by deepfire** [comment]
@gromakovsky, there's still a problem with OpenSSL:
```
Exception calling "DownloadFile" with "2" argument(s): "The remote server returned an error: (404) Not Found."
At line:1 char:101
+ ... eb.com/download/Win64OpenSSL-1_0_2L.exe', "$($env:USERPROFILE)\Win64O ...
+                                                  ~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : WebException
 
Command executed with exception: Exception calling "DownloadFile" with "2" argument(s): "The remote server returned an error: (404) Not Found."
```

**Comment by gromakovsky** [comment]
@deepfire CI passes now. Can you approve if everything is ok?

**Comment by deepfire** [comment]
@dcoutts, can we merge this?

**Comment by deepfire** [comment]
Thank you, Duncan!

**Pull request merged**